src/main/java/carpet/mixins/Explosion_scarpetEventMixin.java

```diff
@@ -0,0 +1,39 @@
+package carpet.mixins;
+
+import net.minecraft.entity.damage.DamageSource;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.explosion.Explosion;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.List;
+
+import static carpet.script.CarpetEventServer.Event.EXPLOSION;
+
+@Mixin(value = Explosion.class)
```

Comment:
You might want to apply proper priorities to make sure ExplosionMixin with optimized explosions is mixed in later (so not cancelled the event)

Reply:
Ehm... I think it's the reverse 😂
Higher values first, then lower values.

In this case I have to mix the optimization first and then inject my event handler at the beginning of the method with the injection - I had to give a lower prioritize to the event-mixin.